prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/command/impl/GatewayThreadImpl.java

```diff
@@ -153,6 +155,41 @@ private Promise<Object> performAction(String model, String provider, String serv
         return d.getPromise();
     }
 
+    private <T> Promise<TimedValue<T>> pullResourceValue(String model, String provider, String service,
```

Comment:
This behaviour should live outside the gateway thread, probably in the model nexus or the resource itself. 
Having reflected, we shouldn't be grabbing the gateway thread's promise factory for the work either as that's really for command use. GET/SET/ACT should share their own thread pool for doing the calls and then update the twin on the gateway thread with a command.